File: api/src/models/supplier.ts
/**
 * @swagger
 * components:
 *   schemas:
 *     Supplier:
 *       type: object
 *       required:
 *         - supplierId
 *         - name
 *       properties:
 *         supplierId:
 *           type: integer
 *           description: The unique identifier for the supplier
 *         name:
 *           type: string
 *           description: The name of the supplier
 *         address:
 *           type: string
 *           description: The physical address of the supplier
 *         contactPerson:
 *           type: string
 *           description: Name of the primary contact person
 *         email:
 *           type: string
 *           format: email
 *           description: Contact email for the supplier
 *         phone:
 *           type: string
 *           description: Contact phone number for the supplier
 *         description:
 *           type: string
 *           description: Additional details about the supplier
 */
export interface Supplier {
  supplierId: number;
  name: string;
  description: string;
  contactPerson: string;
  email: string;
  phone: string;
}

File: api/src/main/java/com/github/av2/api/model/Supplier.java
package com.github.av2.api.model;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Schema(description = "Supplier information")
public class Supplier {
    @Schema(description = "The unique identifier for the supplier", required = true)
    private Integer supplierId;
    
    @Schema(description = "The name of the supplier", required = true)
    private String name;
    
    @Schema(description = "Description of the supplier")
    private String description;
    
    @Schema(description = "Contact person for the supplier")
    private String contactPerson;
    
    @Schema(description = "Contact email for the supplier")
    private String email;
    
    @Schema(description = "Contact phone number for the supplier")
    private String phone;
}